rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java

```diff
@@ -187,22 +191,26 @@ private Account getAccount(String from, String passphrase) {
         return wallet.getAccount(new RskAddress(from), passphrase);
     }
 
-	private String sendTransaction(CallArguments args, Account senderAccount) throws Exception {
+    private String sendTransaction(CallArguments args, Account senderAccount) throws Exception {
 
-		if (senderAccount == null) {
-			throw new Exception("From address private key could not be found in this node");
-		}
+        if (senderAccount == null) {
+            throw new Exception("From address private key could not be found in this node");
+        }
+
+        TransactionArguments txArgs = TransactionArgumentsUtil.processArguments(args, transactionPool, senderAccount, config.getNetworkConstants().getChainId());
 
-		TransactionArguments txArgs = TransactionArgumentsUtil.processArguments(args, transactionPool, senderAccount, config.getNetworkConstants().getChainId());
+        Transaction tx = Transaction.builder().withTransactionArguments(txArgs).build();
 
-		Transaction tx = Transaction.builder().withTransactionArguments(txArgs).build();
+        tx.sign(senderAccount.getEcKey().getPrivKeyBytes());
 
-		tx.sign(senderAccount.getEcKey().getPrivKeyBytes());
+        if (!tx.acceptTransactionSignature(constants.getChainId())) {
+            throw RskJsonRpcRequestException.invalidParamError(TransactionArgumentsUtil.ERR_INVALID_CHAIN_ID + tx.getChainId());
```

Comment:
please cover this thrown exception in unit tests

Reply:
Done